src/main/java/org/jenkinsci/plugins/casc/BaseConfigurator.java

```diff
@@ -53,12 +55,27 @@
                 continue; // read only
             }
             if (setter.getAnnotation(Deprecated.class) != null) {
-                LOGGER.log(Level.FINE, "Ignored {0} property: deprecated", name);
-                continue; // not actually public
+                switch (ConfigurationAsCode.get().getDeprecation()) {
+                    case reject:
+                        LOGGER.log(Level.WARNING, "Ignored {0} property: deprecated", name);
+                        continue; // not actually public
+                    case warn:
+                        LOGGER.log(Level.WARNING, "Ignored {0} property: deprecated", name);
+                }
             }
-            if (setter.getAnnotation(Restricted.class) != null) {
-                LOGGER.log(Level.FINE, "Ignored {0} property: restricted", name);
-                continue; // not actually public     - require access-modifier 1.12
+            final Restricted r = setter.getAnnotation(Restricted.class);
+            if (r != null) {
+                switch (ConfigurationAsCode.get().getRestricted()) {
+                    case beta:
+                        for (Class aClass : r.value()) {
+                            if (aClass == Beta.class) break;
```

Comment:
it will break the cycle, but not the "case" switch.  So it will fall through to `reject` IIRC Java